src/node/module.ts

```diff
@@ -21,16 +24,24 @@ export function createRequire(
     );
   }
 
-  return moduleUtil.createRequire(normalizedPath);
+  // TODO(soon): We should move this to C++ land.
+  // Ref: https://nodejs.org/docs/latest/api/modules.html#requireid
+  return Object.assign(moduleUtil.createRequire(normalizedPath), {
+    // We don't throw ERR_METHOD_NOT_IMPLEMENTED because it's too disruptive.
+    resolve(): void {
+      return undefined;
+    },
+    extensions: undefined,
```

Comment:
That does not seem to adheres to NodeJS types?

Also can a test be added?

Reply:
I personally think it's fine if the types don't match on apis we intentionally do not support and that have been deprecated in node.js. If we want to be strict about it, however, these can return an empty object instead. e.g. `return {};`

Reply:
I'll just remove extensions since it's deprecated and we're have a policy of not implementing deprecated/experimental features.

Reply:
what about resolve?

Reply:
unenv polyfill throws an error afaict. i'm not sure what we consider a "breaking change" at this point. @jasnell

```
resolve: Object.assign(notImplemented("module.require.resolve"), {
      paths: notImplemented("module.require.resolve.paths"),
    }),
    ```

Reply:
If `resolve` in the polyfill exists as a function but throws an error when called, that's what we should do also.

Generally speaking, for all of these, we have to generally align with whatever the polyfill is currently doing. We can tweak from there using compat flags/dates later, but if the goal is to replace the polyfill with the built-in non-op stub, then the behavior of the built-in non-op stub should match the polyfill or we risk breaking users.

Reply:
Sorry I might be wrong, not really have time tonight.

I think it's a breaking change because currently workerd returns a string for `createRequire(root).resolve()` - IIRC @jasnell implemented that.

Also @anonrig I think you are looking at the wrong file, [unenv code](https://github.com/unjs/unenv/blob/main/src/runtime/node/module/%24cloudflare.ts#L64-L72) is

```ts
export const createRequire: typeof nodeModule.createRequire = (
  file: string | URL,
) => {
  return Object.assign(workerdModule.createRequire(file), {
    resolve: Object.assign(notImplemented("module.require.resolve"), {
      paths: notImplemented("module.require.resolve.paths"),
    }),
    cache: Object.create(null),
    extensions: _extensions,
    main: undefined,
  });
};
```

Hope this helps

Reply:
Hmmm, good point about implementnig `createRequire(...).resolve(...)` ... we may be able to implement resolve here reasonably by having it defer to `createRequire(...).resolve(...)` so we don't have to leave this stubbed out and things stay more consistent.

Reply:
> Wait... actually _did I_ implement `createRequire(...).resolve(...)`

Yes in https://github.com/cloudflare/workerd/pull/2636 :)